importlib_metadata/__init__.py

```diff
@@ -696,16 +696,25 @@ def _convert_egg_info_reqs_to_simple_reqs(sections):
         def make_condition(name):
             return name and f'extra == "{name}"'
 
-        def parse_condition(section):
+        def quoted_marker(section):
             section = section or ''
             extra, sep, markers = section.partition(':')
             if extra and markers:
                 markers = f'({markers})'
             conditions = list(filter(None, [markers, make_condition(extra)]))
             return '; ' + ' and '.join(conditions) if conditions else ''
 
+        def url_req_space(req):
+            """
+            PEP 508 requires a space between the url_spec and the quoted_marker.
+            Ref python/importlib_metadata#357.
+            """
+            # '@' is uniquely indicative of a url_req.
+            return ' ' * ('@' in req)
```

Comment:
_Technically_ this is not correct because `@` can be found in a version here; the version format in an `egg-info` is not PEP 508 compliant (it predates the standard). But adding a space is always fine, so no bugs will come out of this.